mod_authz_unixgroup/mod_authz_unixgroup.c

```diff
@@ -112,6 +116,87 @@ static int check_unix_group(request_rec *r, const char *grouplist)
     if (at != NULL) *at= '@';
     return 0;
 }
+#else
+#define MAX_USER_GRPS (4*1024)
+static int check_unix_group(request_rec *r, const char *grouplist)
+{
+    char **p;
+    char *user= r->user;
+    char *w, *at;
+    static gid_t groups[MAX_USER_GRPS], gid;
+    int ngroups = MAX_USER_GRPS, i;
+
+    /* Strip @ sign and anything following it from the username.  Some
+     * authentication modules, like mod_auth_kerb like appending such
+     * stuff to user names, but an @ sign is never legal in a unix login
+     * name, so it should be safe to always discard such stuff.
+     */
+    if ((at= strchr(user, '@')) != NULL) *at= '\0';
```

Comment:
mod_auth_kerb is very old but still functions, if it had @ in the past, I think it still has.

Reply:
What I was trying to say is that our current code (and this PR) just skips the domain (everything after the '@'), which could potentially be a security hole if the system is linked to multiple domains which both have a user with the same name.

I was wondering if the newer functions (such as `getgrouplist`) now support specifying domains (e.g. 'user@domain.com' or 'domain.com\user'), so maybe we can stop skipping the domain?

**But that is beyond the scope of this PR, so we can take a look at it sometime down the road. I think this PR looks good. If there are no comments by next week, I will merge it and make a new release.**

Reply:
> What I was trying to say is that our current code (and this PR) just skips the domain (everything after the '@'), which could potentially be a security hole if the system is linked to multiple domains which both have a user with the same name.
> 
> I was wondering if the newer functions (such as `getgrouplist`) now support specifying domains (e.g. '[user@domain.com](mailto:user@domain.com)' or 'domain.com\user'), so maybe we can stop skipping the domain?
> 
> **But that is beyond the scope of this PR, so we can take a look at it sometime down the road. I think this PR looks good. If there are no comments by next week, I will merge it and make a new release.**

I see it also dangerous to strip the domain by default.

Either add a particular option to strip a particular domain or skip users at all having domain inside by default.

Reply:
> I see it also dangerous to strip the domain by default.
> 
> Either add a particular option to strip a particular domain or skip users at all having domain inside by default.

I agree, I will add a new issue for this problem. Since this PR doesn't change that at all, it should be good to go.

For the future, that idea to strip specific domains is a good one, which will probably be a good solution if testing shows that the new functions (e.g. `getgrouplist`) still don't support domains.

Reply:
@pbiering Issue #57 has been added to address this issue.